addons/binding/org.openhab.binding.autelis/src/main/java/org/openhab/binding/autelis/handler/AutelisHandler.java

```diff
@@ -341,22 +367,66 @@ protected void execute() {
          * This xmlDoc will now contain the three XML documents we retrieved
          * wrapped in response tags for easier querying in XPath.
          */
+        HashMap<String, String> pumps = new HashMap<>();
         String xmlDoc = sb.toString();
         for (Channel channel : getThing().getChannels()) {
-            String key = channel.getUID().getId().replace('-', '/');
+            String key = channel.getUID().getId().replaceFirst("-", "/");
             XPathFactory xpathFactory = XPathFactory.newInstance();
             XPath xpath = xpathFactory.newXPath();
             try {
                 InputSource is = new InputSource(new StringReader(xmlDoc));
-                String value = xpath.evaluate("response/" + key, is);
+                String value = null;
+
+                /**
+                 * Work around for Pentair pumps. Rather then have child XML elements, the response rather uses commas
+                 * on the pump response to separate the different values like so:
+                 *
+                 * watts,rpm,gpm,filter,error
+                 *
+                 * Also, some pool will only report the first 3 out of the 5 values.
+                 */
+
+                Matcher matcher = pumpsPattern.matcher(key);
+                if (matcher.matches()) {
+                    if (!pumps.containsKey(key)) {
+                        String pumpValue = xpath.evaluate("response/" + matcher.group(1), is);
+                        String[] values = pumpValue.split(",");
+                        for (int i = 0; i < PUMP_TYPES.length; i++) {
+
+                            // this will be something like pump/pump1-rpm
+                            String newKey = matcher.group(1) + '-' + PUMP_TYPES[i];
+
+                            // some Pentair models only have the first 3 values
+                            if (i < values.length) {
+                                pumps.put(newKey, values[i]);
+                            } else {
+                                pumps.put(newKey, "");
+                            }
+                        }
+                    }
+                    value = pumps.get(key);
+                } else {
+                    value = xpath.evaluate("response/" + key, is);
+
+                    // Convert pentair salt levels to PPM.
+                    if ("chlor/salt".equals(key)) {
+                        try {
+                            value = String.valueOf(Integer.parseInt(value) * 50);
```

Comment:
As for breaking it out, I'm indifferent there, it would make it more readable, but on the other hand, its only ever used once in the code.  As for the string conversion,  I also realized its inefficient to do this, but on the other hand it would add much more complication to the code and how I'm converting to an item's accepted type.